### PR TITLE
Cleanup and restructure

### PR DESCRIPTION
**Pull request by patrickocoffeyo**
This PR introduces eslint, separates functionality into methods sensibly, and adds JSDoc.

### CODE REVIEW

**Comment by mattgrill** [comment]
Needs a gif.

**Comment by patrickocoffeyo** [comment]
slenuh
kill em with kindness tho

File: index.js [hunk 1]
```diff
@@ -1,43 +1,73 @@
...
+      (error) => {
+        if (error) {
+          console.log(error);
+        }
       }
-    }
-  );
-}
+    );
+  }).catch(console.log);
```
Comment (mattgrill) [inline index.js]: I think you want to do something like, `.catch(err => console.log(err));`

Comment (patrickocoffeyo) [inline index.js:70]: screen shot 2016-12-02 at 2 13 36 pm

**Pull request merged**